homeassistant/components/qwikswitch.py

```diff
@@ -74,129 +77,109 @@ def name(self):
     @property
     def is_on(self):
         """Check if device is on (non-zero)."""
-        return self._qsusb[self._id, 1] > 0
+        return self._qsusb[self.qsid, 1] > 0
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs):
         """Turn the device on."""
         new = kwargs.get(ATTR_BRIGHTNESS, 255)
-        self._qsusb.set_value(self._id, new)
+        self._qsusb.set_value(self.qsid, new)
 
-    @asyncio.coroutine
-    def async_turn_on(self, **kwargs):
-        """Turn the device on."""
-        new = kwargs.get(ATTR_BRIGHTNESS, 255)
-        self._qsusb.set_value(self._id, new)
-
-    def turn_off(self, **kwargs):  # pylint: disable=unused-argument
+    async def async_turn_off(self, **_):
         """Turn the device off."""
-        self._qsusb.set_value(self._id, 0)
-
-    @asyncio.coroutine
-    def async_turn_off(self, **kwargs):  # pylint: disable=unused-argument
-        """Turn the device off."""
-        self._qsusb.set_value(self._id, 0)
-
-
-class QSSwitch(QSToggleEntity, SwitchDevice):
-    """Switch based on a Qwikswitch relay module."""
+        self._qsusb.set_value(self.qsid, 0)
 
-    pass
-
-
-class QSLight(QSToggleEntity, Light):
-    """Light based on a Qwikswitch relay/dimmer module."""
-
-    @property
-    def brightness(self):
-        """Return the brightness of this light (0-255)."""
-        return self._qsusb[self._id, 1] if self._dim else None
-
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        return SUPPORT_BRIGHTNESS if self._dim else None
+    async def async_added_to_hass(self):
+        """Listen for updates from QSUSb via dispatcher."""
+        self.hass.helpers.dispatcher.async_dispatcher_connect(
+            self.qsid, lambda _=None: self.async_schedule_update_ha_state())
```

Comment:
Please swap this lambda for a regular helper method that calls `async_schedule_update_ha_state` and connect that helper method here.